File: CS50/sentimental-readability/readability.py
import re
from cs50 import get_string


def count_letters(text):
    count = 0
    for letter in text:
        if letter.isalpha():
            count += 1
    # print (count)
    return count


def count_words(text):
    # print((text.split(' ')))
    return len(text.split(' '))


def count_sentences(text):
    # print((re.split('[?.!]',text)))
    return len((re.split('[?.!]', text))) - 1
    

def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sen = count_sentences(text)
    index = index = round(0.0588 * letters / words * 100 -
                          0.296 * sen / words * 100 - 15.8)
    print(index)
    if index < 1:
        print("Before Grade 1")
    elif index >= 1 and index <= 16:
        print(f"Grade {index}")
    else:
        print("Grade 16+")


main()

File: CS50/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sen = count_sentences(text);
    // printf("%i, %i ,%i\n",letters,words,sen);
    int index = round(0.0588 * letters / words * 100 - 0.296 * sen / words * 100 - 15.8);
    // printf("%i\n",(int) round(index));
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}



int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (isalpha(text[i]))
        {
            count++ ;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (text[i] == ' ' || text[i] == '\0')
        {
            count++ ;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (text[i] == '.' ||
            text[i] == '?' ||
            text[i] == '!')
        {
            count++ ;
        }
    }
    return count;
}